src/main/java/com/thomashan/coup/PlayerCards.java

```diff
@@ -0,0 +1,65 @@
+package com.thomashan.coup;
+
+
+import com.thomashan.collection.immutable.ImmutableList;
+import com.thomashan.collection.immutable.ImmutableSet;
+
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public final class PlayerCards {
+    private final transient ImmutableList<PlayerCard> playerCards;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [It is somewhat confusing to have a field name matching the declaring class name](https://app.codacy.com/app/thomashan/coup-java/pullRequest?prid=4561385)